### PR TITLE
Tag from the `merge and deploy` command

### PR DESCRIPTION
**Pull request by ReinierMaas**
When calling `merge and deploy` we currently tag the branch from a CI run on the base branch. Instead, we reduce the time it takes from 3 to 2 CI runs by tagging from Hoff. One CI run on the testing branch and one to build the tag.

### CODE REVIEW

**Review by maartenberg — CHANGES_REQUESTED** [review]
As discussed yesterday: our "tag and deploy" machinery ultimately depends on the string "autodeploy" being somewhere in the tag message (automated tags default to "v1234 (autodeploy)").

Could you ensure that tags created for a `merge and deploy` command get "autodeploy" in their message?

File: src/Logic.hs [hunk 1]
```diff
@@ -561,19 +570,29 @@ pushCandidate (pullRequestId, pullRequest) newHead state =
...
           Left (TagName bad) -> do
             commentToUser ("Sorry, I could not tag your PR. The previous tag `" <> bad <> "` seems invalid")
             tryPromote prBranch newHead
           Right v -> do
-            (tagResult, pushResult) <- tryPromoteWithTag prBranch newHead (versionToTag $ v + 1)
+            let previousTag = versionToTag v
+            mChangelog <- getChangelog previousTag newHead
+            let
+              tagName = versionToTag $ v + 1
+              changelog = maybe "no changelog" id mChangelog
```
Comment (maartenberg) [inline src/Logic.hs:586]: Suggestion: AIUI we should only get a `Nothing` here if Hoff fails to get the changelog, not if there is no changelog, in the unlikely case where the previous tag already points to the commit that Hoff tagged I think it should give an empty string as changelog with exit code 0 (see e.g. `git shortlog HEAD..HEAD`).

I suggest making it clearer that something failed:

```suggestion
              changelog = maybe "Failed to get changelog" id mChangelog
```

**Review by maartenberg — APPROVED** [review]
LGTM, thanks for adding this!

I have one minor suggestion, feel free to merge afterwards:

**Comment by ReinierMaas** [comment]
@OpsBotPrime merge!

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @ReinierMaas, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as 3d25cfd71ea93782e7bae90f7d9be8e6ea2b0278, waiting for CI …

**Pull request merged**